opencensus/context/with_context.h

```diff
@@ -21,15 +21,22 @@ namespace opencensus {
 namespace context {
 
 // WithContext is a scoped object that sets the current Context to the given
-// one, until the WithContext object is destroyed.
+// one, until the WithContext object is destroyed. If the condition is false, it
+// doesn't do anything.
 //
-// Because it changes the current (thread local) context, NEVER allocate a
-// WithContext in one thread and deallocate in another. A simple way to ensure
-// this is to only ever stack-allocate it.
+// Because WithContext changes the current (thread local) context, NEVER
+// allocate a WithContext in one thread and deallocate in another. A simple way
+// to ensure this is to only ever stack-allocate it.
+//
+// Example usage:
+// {
+//   WithContext wc(op.ctx_);
+//   // Do work.
+// }
 class WithContext {
  public:
-  explicit WithContext(const Context& ctx);
-  explicit WithContext(Context&& ctx);
+  explicit WithContext(const Context& ctx, bool cond = true);
+  explicit WithContext(Context&& ctx, bool cond = true);
```

Comment:
Does this need to be public?

Reply:
Do you mean WithContext in general? If we want users to be able to capture contexts and install them later without using Wrap() then yes.

Reply:
Bogdan meant the rvalue reference constructor.

The reason for keeping this is: if the caller has saved a copy of a Context, and wants to install it, and won't need it later, they can std::move() it to save another copy.